test/zdtm.py

```diff
@@ -2519,179 +2514,205 @@ def clean_stuff(opts):
-                "--keep-img",
-                help="Whether or not to keep images after test",
-                choices=['always', 'never', 'failed'],
-                default='failed')
-rp.add_argument("--report", help="Generate summary report in directory")
-rp.add_argument("--keep-going",
-                help="Keep running tests in spite of failures",
-                action='store_true')
-rp.add_argument("--ignore-taint",
-                help="Don't care about a non-zero kernel taint flag",
-                action='store_true')
-rp.add_argument("--lazy-pages",
-                help="restore pages on demand",
-                action='store_true')
-rp.add_argument("--lazy-migrate",
-                help="restore pages on demand",
-                action='store_true')
-rp.add_argument("--remote-lazy-pages",
-                help="simulate lazy migration",
-                action='store_true')
-rp.add_argument("--tls", help="use TLS for migration", action='store_true')
-rp.add_argument("--title", help="A test suite title", default="criu")
-rp.add_argument("--show-stats",
-                help="Show criu statistics",
-                action='store_true')
-rp.add_argument("--criu-bin",
-                help="Path to criu binary",
-                default='../criu/criu')
-rp.add_argument("--crit-bin",
-                help="Path to crit binary",
-                default='../crit/crit')
-rp.add_argument("--criu-image-streamer-dir",
-                help="Directory where the criu-image-streamer binary is located",
-                default="../../criu-image-streamer")
-rp.add_argument("--pre-dump-mode",
-                help="Use splice or read mode of pre-dumping",
-                choices=['splice', 'read'],
-                default='splice')
-
-lp = sp.add_parser("list", help="List tests")
-lp.set_defaults(action=list_tests)
-lp.add_argument('-i',
-                '--info',
-                help="Show more info about tests",
-                action='store_true')
-
-gp = sp.add_parser("group", help="Generate groups")
-gp.set_defaults(action=group_tests)
-gp.add_argument("-m", "--max-size", help="Maximum number of tests in group")
-gp.add_argument("-n", "--name", help="Common name for group tests")
-gp.add_argument("-x",
-                "--exclude",
-                help="Exclude tests from --all run",
-                action='append')
-
-cp = sp.add_parser("clean", help="Clean something")
-cp.set_defaults(action=clean_stuff)
-cp.add_argument("what", choices=['nsroot'])
-
-opts = vars(p.parse_args())
-if opts.get('sat', False):
-    opts['keep_img'] = 'always'
-
-if opts['debug']:
-    sys.settrace(traceit)
-
-if opts['action'] == 'run':
-    criu.available()
-for tst in test_classes.values():
-    tst.available()
-
-opts['action'](opts)
-
-for tst in test_classes.values():
-    tst.cleanup()
+if __name__ == '__main__':
+    os.chdir(os.path.dirname(os.path.abspath(__file__)))
+    signal.signal(signal.SIGALRM, alarm)
+    opts = get_cli_args()
```

Comment:
opts isn't used, is it?

Reply:
Actually `opts` has a global scope. It is used in the `main()` function, as well as `finish()`, `wait()`, `wait_all()` methods of `Launcher`, and `check(feature)`, `available()` static methods of the `criu` class, etc.